File: B2929/Main.cpp
#include <bits/stdc++.h>

using namespace std;

int main () {

    int N;
    int valor;
    stack<int> pilhaMenor;
    queue<string> outputs;

    cin >> N;

    while (N--) {

        string textos;
        char c;

        cin >> textos;
        c = textos.at(1);

        if (c == 'U') {  
            cin >> valor;
            if (pilhaMenor.empty())
                pilhaMenor.push(valor);
            else
                pilhaMenor.push(min(pilhaMenor.top(), valor));
        } else if (pilhaMenor.empty()) 
            outputs.push("EMPTY");
          else if (c == 'O') 
            pilhaMenor.pop();                 
          else
            outputs.push(to_string(pilhaMenor.top()));

    }

    N = outputs.size();

    while (N--) {
        cout << outputs.front() << endl;
        outputs.pop();
    }

    return 0;
}

File: B2929/Main.java
package B2929;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws outException {

        int N, valor;
        PilhaEnc pilha = new PilhaEnc();
        PilhaEnc pilhaMenor = new PilhaEnc();
        Scanner sc = new Scanner(System.in);
        FilaEnc outputs = new FilaEnc();
        String string;



        N = Integer.parseInt(sc.nextLine());

        for (int i = 0; i < N; i++) {
          //  if (!string.isEmpty())
          //      string.delete(0, string.length());

            string = (sc.nextLine());

            switch (string.charAt(1)) {
                case 'U':
                    valor = Integer.parseInt(string.substring(5));
                    pilha.push(valor);
                    if (pilhaMenor.vazia())
                        pilhaMenor.push(valor);
                    else if (pilhaMenor.top() >= valor)
                        pilhaMenor.push(valor);
                    break;
                case 'O':
                    if (pilha.vazia())
                        outputs.insere("EMPTY");
                    else {
                        valor = pilha.pop();
                        if (pilhaMenor.top() == valor)
                            pilhaMenor.pop();
                    }
                    break;
                default:
                    if (pilha.vazia())
                        outputs.insere("EMPTY");
                    else
                        outputs.insere(String.valueOf(pilhaMenor.top()));
                    break;


            }



        }


        sc.close();
// bla
        for (int i = 0; i < outputs.tamanho();)
            System.out.println(outputs.remove());

    }

}